react/src/SearchVariants/components/Autocomplete.tsx

```diff
@@ -1,20 +1,112 @@
-import React, { useState } from "react";
-import { CircularProgress, OutlinedInput, useTheme } from "@material-ui/core";
-import { Search } from "@material-ui/icons";
+import React, { useMemo, useState } from "react";
+import {
+    Button,
+    CircularProgress,
+    Divider,
+    InputAdornment,
+    makeStyles,
+    Menu,
+    MenuItem,
+    OutlinedInput,
+    useTheme,
+} from "@material-ui/core";
+import { ArrowDropDown, Check, Search } from "@material-ui/icons";
 import { Autocomplete } from "@material-ui/lab";
+import { snakeCaseToTitle } from "../../functions";
 import { useGenesQuery } from "../../hooks/genes";
 import { GeneAlias } from "../../typings";
 
+type SearchCategory = "gene" | "region" | "variant_position" | "rsld";
```

Comment:
rsid (here and elsewhere, display value would be rsID or refSNP ID, we should check for the exact preference)